### PR TITLE
Rename menu items

### LINKED ISSUE #17: Rename "Newsletter Box" to "Shortcode"
A mentioned in https://github.com/wearerequired/custom-menu-item-types/issues/14#issuecomment-431067124.

### LINKED ISSUE #18: Rename "Line Break" to "Separator"
The "Line Break" type doesn't add a line break, instead it's a `<hr>` element which is more of a separator.

### PR DESCRIPTION
**Pull request by grappler**
fixes #18 fixes #17

### CODE REVIEW

File: inc/Custom_Menu_Items.php [hunk 1]
```diff
@@ -261,12 +263,12 @@ public function nav_menu_item_fields( $nav_menu_item_fields, $context ) {
 				?>
 				<p class="field-shortcode description description-wide">
 					<label for="edit-menu-item-shortcode-<?php echo $context['item']->ID; ?>">
-						<?php _e( 'Shortcode', 'custom-menu-item-types' ); ?><br />
+						<?php _e( 'Shortcode Box', 'custom-menu-item-types' ); ?><br />
```
Comment (ocean90) [inline inc/Custom_Menu_Items.php]:
```suggestion
						<?php _e( 'Shortcode', 'custom-menu-item-types' ); ?><br />
```
The input is for a shortcode not another shortcode box.

File: inc/Custom_Menu_Items.php [hunk 2]
```diff
@@ -243,7 +245,7 @@ public function nav_menu_item_fields( $nav_menu_item_fields, $context ) {
 				$new_nav_menu_item_fields['description'] = $nav_menu_item_fields['description'];
 			}
 
-			if ( 'newsletter_box' === $context['item']->rcmit_type ) {
+			if ( 'shortcode_box' === $context['item']->rcmit_type || 'newsletter_box' === $context['item']->rcmit_type ) {
```
Comment (ocean90) [inline inc/Custom_Menu_Items.php:252]: How about adding this below `if ( ! empty( $context['item']->rcmit_type ) ) {`:

```php
if ( ! empty( $context['item']->rcmit_type ) ) {
    // Migrate old type to new on-the-fly.
	if ( 'newsletter_box' === $context['item']->rcmit_type ) {
		$context['item']->rcmit_type = 'shortcode_box';
    }
```

With that you don't have to check both types here.

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: inc/Custom_Menu_Items.php (hunk 2 above)
Comment (grappler) [reply inc/Custom_Menu_Items.php:252]: In my next PR I am replacing the if statements with a switch case.

Comment (ocean90) [reply inc/Custom_Menu_Items.php:252]: My point is more the migration to the new type which will happen when the menu gets saved.

File: inc/Custom_Menu_Items.php [hunk 3]
```diff
@@ -243,7 +249,7 @@ public function nav_menu_item_fields( $nav_menu_item_fields, $context ) {
...
```
Comment (ocean90) [inline inc/Custom_Menu_Items.php:252]:
```suggestion
			if ( 'shortcode_box' === $context['item']->rcmit_type ) {
```

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: inc/Custom_Menu_Items.php (hunk 3 above)
Comment (grappler) [reply inc/Custom_Menu_Items.php:252]: I will do it in the next pull request.

**Pull request merged**